Development practices/Design patterns.md

```diff
@@ -1,12 +1,74 @@
-- what are design patterns
-- list of design patterns in rails with links to external blogposts
-
-- previous chapters included:
-  - Adapters
-  - Data objects (Value objects)
-  - Decorators
-  - Form objects
-  - Null objects
-  - Policies
-  - Query objects
-  - Service objects
\ No newline at end of file
+
+## Classic design patterns
+
+Design patterns are typical solutions to common problems in software design.
+Each pattern is like a customizable blueprint used to solve a particular design problem in your code.
+They define a language that efficiently improves team communication.
+
+They can also be used to improve code readability and cleanness, as well as to simplify the design.
+
+[Refactoring.guru](https://refactoring.guru) is a great online place to learn more about them.
+
+## Rails patterns
+
+### Factory
+Class that has one creation method with a large conditional.
```

Comment:
A class that creates other classes? :D

Reply:
Just saw this!

I like it somewhat more descriptive :D It technically doesn't only need to return a class, but can also return an object.